I18N-ACTION-890: (for discussion in teleconference) health warning prototypes for normalization

aphillips opened the pull request:
Added a proposed set of health warnings pertaining to Unicode normalization.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/41.html" title="Last updated on Aug 15, 2020, 8:41 PM UTC (e469f11)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/41/cdbf6d5...aphillips:e469f11.html" title="Last updated on Aug 15, 2020, 8:41 PM UTC (e469f11)">Diff</a>

aphillips committed:
I18N-ACTION-890: Added a proposed set of health warnings pertaining to Unicode normalization.

aphillips committed:
Merge branch 'gh-pages' of https://github.com/w3c/bp-i18n-specdev into gh-pages

Perform edits as part of I18N-ACTION-915

aphillips committed:
Address comments from telecon of 2020-06-04.
Add requirement for specifications to pick up the note/warning.

aphillips committed:
Improve the wording of the mustard.

aphillips committed:
Add error and message handling section per: [I18N-ACTION-940]

xfq commented:
Although I think it is better to separate the localization part into a separate PR, but since it is already here, I will comment here.

xfq commented on index.html:
```diff
@@ -1594,6 +1625,25 @@ <h5>Background</h5>
+
+<p class="advisement" id="l10n_api_message_id">APIs and protocols SHOULD provide language independent identifiers for errors. For example, HTTP result codes, such as the familiar <code>404</code> help users communicate which error they received or look up a translation.</p>
+
+<p class="advisement" id="l10n_api_error_message_metadata">APIs and protocols SHOULD include language and direction metadata for all natural language messages, including errors, to ensure proper presentation, even if localization is not provided. See also [[STRING-META]]</p>
```
```suggestion
<p class="advisement" id="l10n_api_error_message_metadata">APIs and protocols SHOULD include language and direction metadata for all natural language messages, including errors, to ensure proper presentation, even if localization is not provided. See also [[STRING-META]].</p>
```

aphillips committed:
Update index.html

Co-authored-by: Fuqiao Xue <xfq@w3.org>

aphillips commented:
> Although I think it is better to separate the localization part into a separate PR, but since it is already here, I will comment here.

The problem is that we never committed the other PR and I didn't realize we still had it uncommitted and open. I intend to agenda+ both this coming week.

aphillips committed:
Address comments from 2020-08-13 teleconference.
- change presentation of normalization note/warning to make clear that these are "quoted"
- move non-normative text outside the mustard in Localization section.

xfq commented:
The font size in the example note is very small... But maybe it's on purpose?

aphillips commented:
@xfq I originally styled it 80%, as Richard suggested in the call, but that was way too small. I also changed from margin of 4em to only adjusting the left/right margins.
